### PR TITLE
auth: change installation example from MySQL to SQLite

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
This avoids one moving part, simplifying the setup.

### Checklist

I have:
- [ ] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: docs/guides/basic-database.rst [hunk 1]
```diff
@@ -1,194 +1,106 @@
...
 
-Remove any earlier :ref:`setting-launch` statements and
-other configuration statements for backends.
+.. code-block:: shell
 
-.. warning::
-  Make sure that you can actually resolve the hostname of
-  your database without accessing the database! It is advised to supply an
-  IP address here to prevent chicken/egg problems!
+    $ sudo apt-get install pdns-backend-sqlite3
```
Comment (pieterlexis) [inline docs/guides/basic-database.rst]:
```suggestion
    sudo apt-get install pdns-backend-sqlite3
```

File: docs/guides/basic-database.rst [hunk 2]
```diff
@@ -1,194 +1,106 @@
...
+.. code-block:: shell
 
-    # /usr/sbin/pdns_server --daemon=no --guardian=no --loglevel=9
-    (...)
-    Dec 30 13:40:09 About to create 3 backend threads for UDP
-    Dec 30 13:40:09 gmysql Connection failed: Unable to connect to database: Access denied for user 'hubert'@'localhost' to database 'pdns-non-existant'
-    Dec 30 13:40:09 Caught an exception instantiating a backend: Unable to launch gmysql connection: Unable to connect to database: Access denied for user 'hubert'@'localhost' to database 'pdns-non-existant'
-    Dec 30 13:40:09 Cleaning up
-    Dec 30 13:40:10 Done launching threads, ready to distribute questions
+    $ sudo yum install pdns-backend-sqlite
```
Comment (pieterlexis) [inline docs/guides/basic-database.rst]:
```suggestion
    sudo yum install pdns-backend-sqlite
```

File: docs/guides/basic-database.rst [hunk 3]
```diff
@@ -1,194 +1,106 @@
...
 
-This is as to be expected - we did not yet add anything to MySQL for
-PowerDNS to read from. At this point you may also see other errors which
-indicate that PowerDNS either could not find your MySQL server or was
-unable to connect to it. Fix these before proceeding.
+This backend is called 'gsqlite3', and needs to be configured in ``pdns.conf``.
```
Comment (pieterlexis) [inline docs/guides/basic-database.rst]:
```suggestion
This backend is called :doc:`'gsqlite3 <../../backends/gsqlite3>'`, and needs to be configured in ``pdns.conf``.
```

File: docs/guides/basic-database.rst [hunk 4]
```diff
@@ -1,194 +1,106 @@
...
+Add the following lines, adjusted for your local setup:
 
-General MySQL knowledge is assumed in this chapter, please do not
-interpret these commands as DBA advice!
+.. code-block:: ini
 
-Example: configuring MySQL
---------------------------
+    launch=gsqlite3
+    gsqlite3-database=/var/lib/powerdns/pdns.sqlite3
```
Comment (pieterlexis) [inline docs/guides/basic-database.rst]: Perhaps a `.. note::` here describing the access permissions of /etc wrt to systemd?

File: docs/guides/basic-database.rst [hunk 5]
```diff
@@ -1,194 +1,106 @@
...
-Please find `the 4.2 schema <https://github.com/PowerDNS/pdns/blob/rel/auth-4.2.x/modules/gmysqlbackend/schema.mysql.sql>`_ and `the 4.1 schema <https://github.com/PowerDNS/pdns/blob/rel/auth-4.1.x/modules/gmysqlbackend/schema.mysql.sql>`_ on GitHub.
+Now create the database (on RPM systems, the schema path is ``/usr/share/doc/pdns-backend-sqlite/schema.sqlite3.sql``):
 
+.. code-block:: shell
 
-.. literalinclude:: ../../modules/gmysqlbackend/schema.mysql.sql
-   :language: SQL
+    $ sudo mkdir /var/lib/powerdns
+    $ sudo sqlite3 /var/lib/powerdns/pdns.sqlite3 < /usr/share/doc/pdns-backend-sqlite3/schema.sqlite3.sql
+    $ sudo chown -R pdns:pdns /var/lib/powerdns
```
Comment (pieterlexis) [inline docs/guides/basic-database.rst]:
```suggestion
    sudo mkdir /var/lib/powerdns
    sudo sqlite3 /var/lib/powerdns/pdns.sqlite3 < /usr/share/doc/pdns-backend-sqlite3/schema.sqlite3.sql
    sudo chown -R pdns:pdns /var/lib/powerdns
```

File: docs/guides/basic-database.rst [hunk 6]
```diff
@@ -1,194 +1,106 @@
...
 
-We recommend you add the following MySQL statements as well. These will add
-foreign key constraints to the tables in order to automate deletion of records, key
-material, and other information upon deletion of a domain from the
-domains table. These will only work on the InnoDB storage engine, but if you
-followed our guide so far, that's exactly the engine we are using.
+And start PowerDNS
```
Comment (pieterlexis) [inline docs/guides/basic-database.rst]:
```suggestion
And (re)start PowerDNS
```

File: docs/guides/basic-database.rst [hunk 7]
```diff
@@ -1,194 +1,106 @@
...
 
-The following SQL does the job:
+.. code-block:: shell
 
-.. literalinclude:: ../../modules/gmysqlbackend/enable-foreign-keys.mysql.sql
+    $ sudo systemctl start pdns
```
Comment (pieterlexis) [inline docs/guides/basic-database.rst]:
```suggestion
    sudo systemctl start pdns
```

File: docs/guides/basic-database.rst [hunk 8]
```diff
@@ -1,194 +1,106 @@
...
 
+or
 
-Now we have a database and an empty table. PowerDNS should now be able
-to launch in monitor mode and display no errors:
+.. code-block:: shell
 
-::
+    $ sudo systemctl restart pdns
```
Comment (pieterlexis) [inline docs/guides/basic-database.rst]:
```suggestion
    sudo systemctl restart pdns
```

File: docs/guides/basic-database.rst [hunk 9]
```diff
@@ -1,194 +1,106 @@
...
-    INSERT INTO records (domain_id, name, content, type,ttl,prio)
-    VALUES (1,'example.com','mail.example.com','MX',120,25);
+.. code-block:: shell
 
-.. warning::
-  Host names and the MNAME of a :ref:`types-soa`
-  records are NEVER terminated with a '.' in PowerDNS storage! If a
-  trailing '.' is present it will inevitably cause problems, problems that
-  may be hard to debug.
+    $  create-zone example.org ns1.example.com
```
Comment (pieterlexis) [inline docs/guides/basic-database.rst]: won't work without sudo, unless your running this as `pdns`

```suggestion
    sudo -u pdns create-zone example.org ns1.example.com
```

File: docs/guides/basic-database.rst [hunk 10]
```diff
@@ -1,194 +1,106 @@
...
+    Creating empty zone 'example.com'
+    Also adding one NS record
+    $ pdnsutil add-record example.com '' MX '25 mail.example.com'
```
Comment (pieterlexis) [inline docs/guides/basic-database.rst]:
```suggestion
    sudo -u pdns pdnsutil add-record example.com '' MX '25 mail.example.com'
```

File: docs/guides/basic-database.rst [hunk 11]
```diff
@@ -1,194 +1,106 @@
...
+    New rrset:
+    example.com. 3005 IN MX 25 mail.example.com
+    $ pdnsutil add-record example.com. www A 192.0.2.1
```
Comment (pieterlexis) [inline docs/guides/basic-database.rst]:
```suggestion
    sudo -u pdns pdnsutil add-record example.com. www A 192.0.2.1
```

File: docs/guides/basic-database.rst [hunk 12]
```diff
@@ -1,194 +1,106 @@
...
 
-    Mar 01 16:04:42 Remote 127.0.0.1 wants 'www.example.com|A', do = 0, bufsize = 1680: packetcache MISS
+.. warning::
+  When debugging DNS problems, don't use ``host``. Please use ``dig`` or ``drill``.
 
-Now we need to add some records to our database (in a separate shell):
+Note the ``REFUSED`` status - this is the code most name servers use to indicate they do not know about a domain.
 
-::
+Now, let's add a zone and some records:
```
Comment (pieterlexis) [inline docs/guides/basic-database.rst]: A paragraph introducing `pdnsutil`, and why raw SQL shouldn't be used would be nice here.

File: docs/guides/basic-database.rst [hunk 13]
```diff
@@ -1,194 +1,106 @@
...
-    * pdns.service - PowerDNS Authoritative Server
-       Loaded: loaded (/lib/systemd/system/pdns.service; enabled)
-       Active: active (running) since Tue 2017-01-17 15:59:28 UTC; 1 months 12 days ago
-         Docs: man:pdns_server(1)
-               man:pdns_control(1)
-               https://doc.powerdns.com
-     Main PID: 24636 (pdns_server)
-       CGroup: /system.slice/pdns.service
-               `-24636 /usr/sbin/pdns_server --guardian=no --daemon=no --disable-syslog --write-pid=no
+Now, run ``pdnsutil edit-zone example.com`` and try to add a few more records, and query them with dig to make sure they work.
```
Comment (pieterlexis) [inline docs/guides/basic-database.rst]: Mention the record and packet caches here?

**Comment by Habbie** [comment]
External review note: the FAQ about 'port in use' and other things on this page, would be way more useful in installation.html

**Comment by pieterlexis** [comment]
The RFC 2136 docs are also full of SQL, it would be nice to change that to pdnsutil commands as well ;)

File: docs/guides/basic-database.rst (hunk 4 above)
Comment (Habbie) [reply docs/guides/basic-database.rst]: Why?

File: docs/guides/basic-database.rst (hunk 10 above)
Comment (Habbie) [reply docs/guides/basic-database.rst]: I feel removing the `$` here makes the block hard to read. I now also have doubts about the other suggestions I committed for that :)

**Comment by Habbie** [comment]
> The RFC 2136 docs are also full of SQL, it would be nice to change that to pdnsutil commands as well ;)

I'd like to separate that work from this PR :)

**Comment by pieterlexis** [comment]
I've rebased this and fixed the comments (mostly)

**Pull request merged**